### PR TITLE
build: Update go modules to build with go 1.18 and support go 1.19

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
This is an update from make go.mod.update to ensure we are consuming the latest stable packages available.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/file/subvolumegroup/controller.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (r *ReconcileCephFilesystemSubVolumeGroup) updateClusterConfig(cephFilesyst
 	// If the cluster has Multus enabled we need to append the network namespace of the driver's
 	// holder DaemonSet in the csi configmap
 	if cephCluster.Spec.Network.IsMultus() {
-		netNamespaceFilePath, err := csi.GenerateNetNamespaceFilePath(r.opManagerContext, r.client, cephCluster.ClusterName, r.opConfig.OperatorNamespace, csi.CephFSDriverShortName)
+		netNamespaceFilePath, err := csi.GenerateNetNamespaceFilePath(r.opManagerContext, r.client, cephCluster.Name, r.opConfig.OperatorNamespace, csi.CephFSDriverShortName)
```
Comment (travisn) [inline pkg/operator/ceph/file/subvolumegroup/controller.go:279]: @leseb The ClusterName field is no longer available. This net namespace file path is intended to be unique across K8s clusters using the same nodes? Or is unique across ceph clusters sufficient and we can just use the namespace? Will it affect upgrades if this value changes from ClusterName to namespace or something else?

**Review by BlaineEXE — APPROVED** [review]
A lot of churn, but it looks good. I really just see one comment that looks like it could be rendered a little funny. And I think it's still good to get Seb's input on your question.

**Comment by travisn** [comment]
Oops, typo on the force push deleted the branch! Who knew?
```
git push origin go-mod-update -df
```
Opening a new PR...

File: pkg/operator/test/volumes.go [hunk 2]
```diff
@@ -145,9 +145,12 @@ type VolumesAndMountsTestDefinition struct {
...
 // TestMountsMatchVolumes tests two things:
 // (1) That each volume mount in each every MountsSpec has a corresponding volume to source it
-//     in the VolumesSpec
+//
+//	in the VolumesSpec
+//
 // (2) That there are no extraneous volumes defined in the VolumesSpec that do not have a
-//     corresponding volume mount in any of the MountsSpec items
+//
+//	corresponding volume mount in any of the MountsSpec items
```
[the remark this answers is not in the document]
Comment (travisn) [reply pkg/operator/test/volumes.go:153]: Yeah will make that update so we don't get this strange auto-formatting.

**Pull request closed without merging**